pandas/core/internals/array_manager.py

```diff
@@ -694,11 +694,27 @@ def _equal_values(self, other) -> bool:
         Used in .equals defined in base class. Only check the column values
         assuming shape and indexes have already been checked.
         """
-        for left, right in zip(self.arrays, other.arrays):
+        from pandas.core.internals import BlockManager  # to avoid circular import
```

Comment:
how about just checking `not isinstance(other, ArrayManager)` with a comment that this implies BlockManager